doc/pantry.md

```diff
@@ -0,0 +1,482 @@
+    size: 7376
+    sha256: ac2601c49cf7bc0f5d66b2793eddc8352f51a6ee989980827a0d0d8169700a03
+name: my-snapshot
+hidden:
+  warp: false
+  wai: true
+drop-packages:
+- wai-extra
+compiler: ghc-8.0.1
+resolver:
+  size: 515969
+  url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/8/21.yaml
+  sha256: 2ec73d520d3e55cb753eaca11a72a9ce95bd9ba7ccaf16de1150d0130a50a5a1
+```
+
+### Overriding the compiler
+
+The following snapshot specification will be identical to `lts-7.1`, but instead
+use `ghc-7.10.3` instead of `ghc-8.0.1`:
+
+```yaml
+resolver: lts-7.1
+compiler: ghc-7.10.3
+```
+
+### Dropping packages
+
+The following snapshot specification will be identical to `lts-7.1`, but without
+the `text` package in our snapshot. Removing this package will cause all the
+packages that depend on `text` to be unbuildable, but they will still be present
+in the snapshot.
+
+```yaml
+resolver: lts-7.1
+drop-packages:
+  - text
+```
+
+### Specifying ghc-options
+
+In order to specify ghc-options for a package, you use the same syntax as the
+[ghc-options](yaml_configuration.md#ghc-options) field for build configuration.
+The following snapshot specification will be identical to `lts-7.1`, but
+provides `-O1` as a ghc-option for `text`:
+
+```yaml
+resolver: lts-7.1
+packages:
+  - text-1.2.2.1
+ghc-options:
+  text: -O1
+```
+
+This works somewhat differently than the stack.yaml `ghc-options` field, in that
+options can only be specified for packages that are mentioned in the custom
+snapshot's `packages` list. It sets the ghc-options, rather than extending those
+specified in the snapshot being extended.
+
+Another difference is that the `*` entry for `ghc-options` applies to all
+packages in the `packages` list, rather than all packages in the snapshot.
+
+### Specifying flags
+
+In order to specify flags for a package, you use the same syntax as the
+[flags](yaml_configuration.md#flags) field for build configuration. The
+following snapshot specification will be identical to `lts-7.1`, but
+it enables the `developer` cabal flag:
+
+```yaml
+resolver: lts-7.1
+packages:
+  - text-1.2.2.1
+flags:
+  text:
+    developer: true
+```
+
+## Freeze command details
+
+To make builds reproducible it makes sense to pin project dependencies to some
+exact versions and this is what is stack's `freeze` command is about.
+
+### Project freezing
+
+The default mode of its invocation:
+
+```
+$ stack freeze
+```
+freezes the following fields from the project's `stack.yaml`
+
+* packages in `extra-deps` which do not include sha256 of their cabal files and
+  which do not specify pantry tree pointer of the package archive
+* `resolver` if it references a remote snapshot and if it does not specify
+  pantry tree pointer of its contents
+
+The command outputs to standard output new project's `stack.yaml` with these
+changes included.
+
+If a project is specified precisely enough stack tells about it and exits.
+
+### Snapshot freezing
```

Comment:
This section is partially duplicated (`--snapshot` is introduced earlier), and has numerous typos and ambiguities. Perhaps just move the (fixed) last paragraph to the first description of snapshot freezing?

Reply:
Hopefully addressed in 01f9e84f2cd76d2e5dd66ba2411a3ba8fac08ca9, thanks.

Reply:
Yes, it is.